spec/controllers/v3/branches_controller_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V3::BranchesController do
+  subject { create :repository_compound }
+  let!(:user) { subject.owner }
+
+  context 'successful' do
+    describe 'action: index' do
+      before do
+        get :index, params: {organizational_unit_id: user.to_param,
+                             repository_id:
+                             subject.to_param.sub(%r{\A[^/]*/}, '')}
+      end
+      it { expect(response).to have_http_status(:ok) }
+      it { |example| expect([example, response]).to comply_with_rest_api }
+      it 'finds the repository' do
+        expect(response_data['repository']).not_to be(nil)
+      end
+    end
+
+    describe 'action: show' do
+      before do
+        get :show, params: {organizational_unit_id: user.to_param,
+                            repository_id:
+                            subject.to_param.sub(%r{\A[^/]*/}, ''),
+                            name: subject.git.default_branch}
+      end
+      it { expect(response).to have_http_status(:ok) }
+      it { |example| expect([example, response]).to comply_with_rest_api }
+      it 'finds the branch' do
+        expect(response_data['repository']['branch']).not_to be(nil)
+      end
```

Comment:
You could replace this a test that checks if the correct branch has been returned:
```ruby
it 'finds the correct branch' do
  expect(response_data['repository']['branch']['name']).to eq(repository.git.default_branch)
end
```
The not-nil property is checked implicitly with this.